geode-core/src/main/java/org/apache/geode/cache/wan/GatewaySender.java

```diff
@@ -178,8 +178,11 @@
   /**
    * Starts this GatewaySender. Once the GatewaySender is running, its configuration cannot be
    * changed.
+   * In case, GatewaySender was previosly stopped, and some events wefe leftin queue, with option
+   * cleanQueues
+   * we choose whether to restore or discard previos queue. Default is to restore queue.
    */
-  void start();
+  void start(boolean cleanQueues);
```

Comment:
This is a public interface, adding this parameter here would break backward compatibility, which is not "allowed" unless we're releasing a major version of the product. "Common" users generally don't implement this interface at all, fair enough, but there are some frameworks (like `spring-data-geoge`) that do (see [here](https://github.com/spring-projects/spring-data-geode/blob/master/spring-data-geode/src/main/java/org/springframework/data/gemfire/wan/GatewaySenderWrapper.java#L34)), and they will be in trouble if we break the backward compatibility.

Reply:
Thanks for comment.